hilti/toolchain/src/compiler/validator.cc

```diff
@@ -196,27 +196,33 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
             if ( nodes.size() <= 1 )
                 continue;
 
-            const auto& first_node = *std::min_element(nodes.begin(), nodes.end(), [](const auto& a, const auto& b) {
-                return a->location() < b->location();
-            });
+            auto sorted_nodes = std::vector<Declaration*>(nodes.begin(), nodes.end());
+            std::sort(sorted_nodes.begin(), sorted_nodes.end(),
+                      [](const auto& a, const auto& b) { return a->location() < b->location(); });
 
-            for ( const auto& node : nodes ) {
+            const auto& first_node = *sorted_nodes.begin();
+            for ( std::size_t i = 1; i < sorted_nodes.size(); i++ ) {
+                const auto& node = sorted_nodes[i];
                 if ( node->location() == first_node->location() )
                     continue;
```

Comment:
We seem to assume below that all nodes have different locations, so with the iteration starting at `1` here this should be technically dead code.

```suggestion
```